### PR TITLE
runtime: match Go complex division semantics

### PR DESCRIPTION
**Pull request by cpunion**
## Summary
- Lower complex division through a runtime helper that matches Go special-value semantics.
- Cover division with infinite, zero, and max finite components in `./test/go`.
- Update the focused complex IR source check for the helper call.

## Example
From `TestComplexDivisionSpecialValues`:

```go
num := complex(1.0, 1.0)
den := complex(math.Inf(1), math.Inf(1))
got := num / den
```

Go has specific complex division semantics around infinities and NaNs. LLGO must produce the same finite/zero/NaN results as Go for these edge cases.

## Without This PR
LLGO lowers complex division directly in a way that can produce `NaN+NaNi` for cases where Go produces a defined special-value result. The regression test fails before the fix.

## Verification
- `./dev/llgo.sh test ./test/go -run TestComplexDivisionSpecialValues -count=1`
- `./dev/llgo.sh test ./test/go -run TestComplexDivision -count=1`
- `./dev/llgo.sh test ./test/go -count=1`
- `go test ./ssa -run TestFromTestrt/complex$ -count=1`
- `go test ./ssa -count=1`

### CODE REVIEW

File: runtime/internal/runtime/z_complex.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+func complexDivInf2one(x float64) float64 {
+	r := 0.0
+	if complexDivIsInf(x) {
+		r = 1.0
+	}
+	return complexDivCopysign(r, x)
+}
+
+func Complex128Div(n, m complex128) complex128 {
```
[the remark this answers is not in the document]
Comment (cpunion) [reply runtime/internal/runtime/z_complex.go:59]: This is a fair readability suggestion. I’m keeping this PR focused on the semantic fix, but I agree a short doc comment would help if we touch this file again.

File: runtime/internal/runtime/z_complex.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+func complexDivIsFinite(x float64) bool {
+	return !complexDivIsNaN(x) && !complexDivIsInf(x)
+}
+
+func complexDivInf() float64 {
+	return 1 / complexDivZero
+}
+
+func complexDivCopysign(x, y float64) float64 {
+	if y < 0 || y == 0 && 1/y < 0 {
```
[the remark this answers is not in the document]
Comment (cpunion) [reply runtime/internal/runtime/z_complex.go:45]: The precedence here is intentional and the condition is already evaluated correctly. I’m leaving it as-is for now to keep the change minimal.

File: runtime/internal/runtime/z_complex.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			f = complexDivCopysign(inf, c) * b
+		case (complexDivIsInf(a) || complexDivIsInf(b)) && complexDivIsFinite(c) && complexDivIsFinite(d):
+			a = complexDivInf2one(a)
+			b = complexDivInf2one(b)
+			e = inf * (a*c + b*d)
+			f = inf * (b*c - a*d)
+		case (complexDivIsInf(c) || complexDivIsInf(d)) && complexDivIsFinite(a) && complexDivIsFinite(b):
+			c = complexDivInf2one(c)
+			d = complexDivInf2one(d)
+			e = 0 * (a*c + b*d)
```
[the remark this answers is not in the document]
Comment (cpunion) [reply runtime/internal/runtime/z_complex.go:90]: Agreed that the `0 * expr` path is subtle. The behavior is intentional for NaN propagation in the recovery branch; I’m not changing it in this PR, but the rationale is exactly that.

File: runtime/internal/runtime/z_complex.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package runtime
+
+var complexDivZero float64
```
[the remark this answers is not in the document]
Comment (cpunion) [reply runtime/internal/runtime/z_complex.go:19]: Yes, `complexDivZero` is only there to synthesize `+Inf` inside runtime code without importing `math`. I’m keeping the implementation minimal here, but that is the intent.

File: test/go/binop_test.go [hunk 5]
```diff
@@ -231,6 +232,31 @@ func TestComplexDivisionAndMultiplication(t *testing.T) {
...
+		},
+		{
+			name: "finite over max finite denominator",
+			num:  complex(1, 1),
+			den:  complex(math.MaxFloat64, math.MaxFloat64),
+		},
+	}
+	for _, tc := range cases {
+		got := tc.num / tc.den
+		if real(got) != 0 || imag(got) != 0 {
```
[the remark this answers is not in the document]
Comment (cpunion) [reply test/go/binop_test.go:254]: The extra cases make sense as follow-up coverage, but the current PR was scoped to fixing the incorrect lowering and validating the newly introduced recovery path. I’m leaving broader matrix expansion out of this change.

File: ssa/expr.go [hunk 6]
```diff
@@ -488,29 +488,10 @@ func (b Builder) BinOp(op token.Token, x, y Expr) Expr {
...
-							llvm.CreateBinOp(b.impl, llvm.FMul, xi, yr),
-							llvm.CreateBinOp(b.impl, llvm.FMul, xr, yi),
-						),
-						d,
-					),
-				)
-				return b.aggregateValue(x.Type, r, i)
+				x128 := b.Convert(b.Prog.Complex128(), x)
+				y128 := b.Convert(b.Prog.Complex128(), y)
+				ret := b.InlineCall(b.Pkg.rtFunc("Complex128Div"), x128, y128)
```
[the remark this answers is not in the document]
Comment (cpunion) [reply ssa/expr.go:493]: Ack. This is a real call today, not actual inlining. That tradeoff is intentional here to get correctness aligned first.

File: runtime/internal/runtime/z_complex.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+func complexDivAbs(x float64) float64 {
+	if x < 0 {
+		return -x
+	}
+	return x
+}
```
[the remark this answers is not in the document]
Comment (cpunion) [reply runtime/internal/runtime/z_complex.go:26]: `complexDivAbs` is only used for relative-magnitude branching here. `-0.0` and `+0.0` compare the same for that purpose, so the signed-zero distinction does not affect this algorithm in its current usage.
